datalad/distribution/create.py

```diff
@@ -10,78 +10,205 @@
+
+from .dataset import Dataset
+from .dataset import datasetmethod
+from .dataset import EnsureDataset
+
+
+__docformat__ = 'restructuredtext'
 
 lgr = logging.getLogger('datalad.distribution.create')
 
 
 class Create(Interface):
-    """Create a new dataset.
+    """Create a new dataset from scratch.
+
+    This command initializes a new :term:`dataset` at a given location, or the
+    current directory. The new dataset can optionally be registered in an
+    existing :term:`superdataset` (the new dataset's path needs to be located
+    within the superdataset for that, and the superdataset will be detected
+    automatically). It is recommended to provide a brief description to label
+    the dataset's nature *and* location, e.g. "Michael's music on black
+    laptop". This helps humans to identify data locations in distributed
+    scenarios.  By default an identifier comprised of user and machine name,
+    plus path will be generated.
+
+    Plain Git repositories can be created via the [PY: `no_annex` PY][CMD: --no-annex CMD] flag.
+    However, the result will not be a full dataset, and, consequently,
+    not all features are supported (e.g. a description).
+
+    || REFLOW >>
+    To create a local version of a remote dataset use the
+    :func:`~datalad.api.install` command instead.
+    << REFLOW ||
 
-    This command initializes a new repository at a given location, or the
-    current directory.
+    .. note::
+      Power-user info: This command uses :command:`git init`, and
+      :command:`git annex init` to prepare the new dataset. Registering to a
+      superdataset is performed via a :command:`git submodule add` operation
+      in the discovered superdataset.
     """
 
     _params_ = dict(
-        loc=Parameter(
-            args=("loc",),
-            doc="""location where the dataset shall be  created.  If `None`,
-            is given a dataset will be created in the current working
-            directory""",
+        path=Parameter(
+            args=("path",),
+            metavar='PATH',
+            doc="""path where the dataset shall be created, directories
+            will be created as necessary. If no location is provided, a dataset
+            will be created in the current working directory. Either way the
+            command will error if the target directory is not empty.
+            Use `force` to create a dataset in a non-empty directory.""",
             nargs='?',
             # put dataset 2nd to avoid useless conversion
             constraints=EnsureStr() | EnsureDataset() | EnsureNone()),
-        description=Parameter(
-            args=("-D", "--description",),
-            doc="""short description that humans can use to identify the
-            repository/location, e.g. "Precious data on my laptop."""""),
+        force=Parameter(
+            args=("-f", "--force",),
+            doc="""enforce creation of a dataset in a non-empty directory""",
+            action='store_true'),
```

Comment:
Constraints are not checked in Python, only used for docs. We cannot pass a constraint to argparse in case of `action="store_*"`. Argparse will refuse to work.

I fear that if we enforce constraints within Python, we will suffer from their limitations more than it may (not sure even) help with the case above. The constraints are really designed to convert a string to some proper data type -- not an arbitrary type into something else. In the general case, the "just try conversion in order of specification" approach would quickly not be able to capture the necessary procedures. IMHO.